miden/tests/integration/operations/stack_ops.rs

```diff
@@ -14,17 +197,462 @@ fn swapdw() {
+        let asm_op = "drop";
+        let mut expected_values = test_values.to_vec();
+        expected_values.remove(MIN_STACK_DEPTH-1);
+        expected_values.reverse();
+        expected_values.push(0);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dropw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "dropw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.truncate(MIN_STACK_DEPTH-WORD_LEN);
+        expected_values.reverse();
+        expected_values.append(&mut vec![0; WORD_LEN]);
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn padw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "padw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..WORD_LEN);
+        expected_values.append(&mut vec![0; WORD_LEN]);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dup_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "dup";
+        let mut expected_values = test_values.to_vec();
+        expected_values.remove(0);
+        expected_values.push(*expected_values.last().unwrap());
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 0_usize..MIN_STACK_DEPTH) {
+        let asm_op = format!("dup.{n}");
+        let mut expected_values = test_values.to_vec();
+        let a = expected_values[MIN_STACK_DEPTH-n-1];
+        expected_values.remove(0);
+        expected_values.push(a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupw_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH)) {
+        let asm_op = "dupw";
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..WORD_LEN);
+        let mut a = test_values[MIN_STACK_DEPTH-WORD_LEN..].to_vec();
+        expected_values.append(&mut a);
+        expected_values.reverse();
+        build_op_test!(asm_op, &test_values).prop_expect_stack(&expected_values)?;
+    }
+
+    #[test]
+    fn dupwn_proptest(test_values in prop::collection::vec(any::<u64>(), MIN_STACK_DEPTH), n in 0_usize..WORD_LEN) {
+        let asm_op = format!("dupw.{n}");
+        let mut expected_values = test_values.to_vec();
+        expected_values.drain(0..WORD_LEN);
+        let mut a = test_values[MIN_STACK_DEPTH-WORD_LEN*(n+1)..MIN_STACK_DEPTH-WORD_LEN*n].to_vec();
```

Comment:
This is a bit hard to read. I think something like the following would be easier:
```
        let start_dupw = MIN_STACK_DEPTH - WORD_LEN * (n + 1);
        let end_dupw = MIN_STACK_DEPTH - WORD_LEN * n;
        let mut a = test_values[start_dupw..end_dupw].to_vec();
```